pkg/p2p/libp2p/libp2p.go

```diff
@@ -176,11 +176,7 @@ func New(ctx context.Context, signer beecrypto.Signer, networkID uint64, overlay
 	// If you want to help other peers to figure out if they are behind
 	// NATs, you can launch the server-side of AutoNAT too (AutoRelay
 	// already runs the client)
-	if _, err = autonat.NewAutoNATService(ctx, h,
-		// Support same non default security and transport options as
-		// original host.
-		append(transports, security)...,
-	); err != nil {
+	if _, err = autonat.New(ctx, h); err != nil {
```

Comment:
This change seem to completely remove AutoNAT service.

If we were to update library version, while keeping "existing" behavior, we would maybe have to have change like this:

```go
// Support same non default security and transport options as
// original host.
dialer, err := libp2p.New(ctx, append(transports, security)...)
if err != nil {
	return nil, err
}

// If you want to help other peers to figure out if they are behind
// NATs, you can launch the server-side of AutoNAT too (AutoRelay
// already runs the client)
if _, err = autonat.New(ctx, h, autonat.EnableService(dialer.Network())); err != nil {
	return nil, fmt.Errorf("autonat: %w", err)
}
```

Reply:
Thanks @zbiljic for pointing this out today. This indeed seems like the right solution, however I am wondering if we need a clean way to close the autonat instance and whether we should perform cleanups on shutdown on that `dialer` libp2p instance. I wonder, if this has to be a libp2p instance at all...

Reply:
I pushed these changes @zbiljic